server/datastore/mysql/apple_mdm.go

```diff
@@ -2173,10 +2162,17 @@ func (ds *Datastore) bulkSetPendingMDMAppleHostProfilesDB(
 		updatedDB = true
 	}
 
+	// Define mechanisms for batching our upserts and assemble the payloads for the
+	// profiles we want to insert or update.
 	profilesToInsert := make(map[string]*fleet.MDMAppleProfilePayload)
 
 	executeUpsertBatch := func(valuePart string, args []any) error {
 		// Check if the update needs to be done at all.
+		//
+		// TODO(sarah): What are we gaining from this select step vs. relying on the
+		// profileIntersection to avoid unnecessary updates? I suppose there may be some
+		// cases where intervening changes arrive after we pulled the data for the intersection,
+		// but in that case maybe we are ok with just letting ON DUPLICATE KEY UPDATE handle it.
```

Comment:
Yes, I think this code can be removed. If we called this method, then `updatedDB` should always be set to true (for activity feed)